### DISCUSSION TITLE
Auto label the files in a project that are commonplace on GitHub versus likely unique by the author

### DISCUSSION THREAD

**Discussion by galonpy** [community/community#40881] [Repositories]
### Select Topic Area

Product Feedback

### Body

I have worked on several projects where someone else has put their repo on github but its not clear what part of the work they are the experts on or actually created.  Many people upload a ton of files on their first upload of which many are just generic and downloaded from elsewhere without a reference in the comments of the source.
As a non-expert coming into a new project, it would save a lot of time to know where the unique contributions of that project are. Those unique contributions are not full files that generally exist already.

It should not be hard for GitHub to develop a search engine feature that looks for an exact replica of a file already existing elsewhere on GitHub and to then autolabel when that is true (and then offer a  link to the match).

**For example**, i forked my friend's project here https://github.com/galonpy/newsapi, and I know that he didnt code every single file since he is a beginner programmer. I am not an expert in this area of full stack engineering, but I do have general knowledge of programming and I know the general intent of his project. It would be extremely useful to have automatic labels on which files are unique versus completely copied from elsewhere. This would save me from asking dumb questions when I could get the answer automatically. 

Additional: Even the structure of most of the names and hierarchy of the folders could probably be easily attributed by a GitHub search algorithm to a common source.

**Comment by ossanna16** [comment]
Hi there @galonpy and welcome to our community! Thank you for the feedback!

**Reply by galonpy** [reply]
Thanks @ossanna16 ! Can I get your thoughts?

**Reply by ossanna16** [reply]
Hey @galonpy! I have some personal thoughts but I'm not a subject matter expert on this topic GitHub wise, however I can assure you that our Product Teams are keeping a close eye on this space and read every feature request.
